source/common/router/router.cc

```diff
@@ -100,19 +101,19 @@ void FilterUtility::setUpstreamScheme(Http::HeaderMap& headers, bool use_secure_
   }
 }
 
-bool FilterUtility::shouldShadow(const ShadowPolicy& policy, Runtime::Loader& runtime,
+bool FilterUtility::shouldShadow(const ShadowPolicyPtr& policy, Runtime::Loader& runtime,
```

Comment:
Thinking about this some more I think it might make more sense to deref the pointer instead of passing it to this function, i.e. have this function continue to take a const ref instead of a pointer. That way you avoid having to use pointer indirection in this function.

Reply:
If the idea here is to prefer value-semantics, makes sense to me. So in between 556/557 deref the pointer to a variable, and pass it to proceeding 2 two uses?

Reply:
(updated as described)